packages/kbn-esql-utils/src/utils/run_query.ts

```diff
@@ -21,32 +22,77 @@ export async function getESQLQueryColumns({
   search: ISearchGeneric;
   signal?: AbortSignal;
 }): Promise<DatatableColumn[]> {
-  const response = await lastValueFrom(
-    search(
-      {
-        params: {
-          query: `${esqlQuery} | limit 0`,
-          version: ESQL_LATEST_VERSION,
+  try {
+    const response = await lastValueFrom(
+      search(
+        {
+          params: {
+            query: `${esqlQuery} | limit 0`,
+            version: ESQL_LATEST_VERSION,
+          },
         },
-      },
-      {
-        abortSignal: signal,
-        strategy: 'esql_async',
-      }
-    )
-  );
+        {
+          abortSignal: signal,
+          strategy: 'esql_async',
+        }
+      )
+    );
 
-  const columns =
-    (response.rawResponse as unknown as ESQLSearchReponse).columns?.map(({ name, type }) => {
-      const kibanaType = esFieldTypeToKibanaFieldType(type);
-      const column = {
-        id: name,
-        name,
-        meta: { type: kibanaType, esType: type },
-      } as DatatableColumn;
+    const columns =
+      (response.rawResponse as unknown as ESQLSearchReponse).columns?.map(({ name, type }) => {
+        const kibanaType = esFieldTypeToKibanaFieldType(type);
+        const column = {
+          id: name,
+          name,
+          meta: { type: kibanaType, esType: type },
+        } as DatatableColumn;
 
-      return column;
-    }) ?? [];
+        return column;
+      }) ?? [];
 
-  return columns;
+    return columns;
+  } catch (error) {
+    throw new Error(
+      i18n.translate('esqlUtils.columnsErrorMsg', {
+        defaultMessage: 'Unable to load columns. {errorMessage}',
+        values: { errorMessage: error.message },
+      })
+    );
+  }
+}
+
+export async function getESQLQueryColumnsRaw({
```

Comment:
nit: So, I guess the only difference here is not converting to Kibana types? I wonder if this could be more clear (e.g. `getESQLQueryColumnsESTypes` or maybe just a function comment that will show up in the IDE.